torchci/lib/benchmark/compilerUtils.ts

```diff
@@ -222,6 +222,84 @@ export function computeGeomean(
   return returnedGeomean;
 }
 
+export function computeExecutionTime(
+  data: CompilerPerformanceData[],
+  passingModels: { [k: string]: any }
+) {
+  const executionTime: { [k: string]: any } = {};
+  const returnedExecutionTime: any[] = [];
+
+  data.forEach((record: CompilerPerformanceData) => {
+    const bucket = record.granularity_bucket;
+    const workflowId = record.workflow_id;
+    const suite = record.suite;
+    const model = record.name;
+    const absLatency = record.abs_latency;
+
+    // Use clear compiler name to avoid confusion about what they do
+    const compiler =
+      COMPILER_NAMES_TO_DISPLAY_NAMES[record.compiler] ?? record.compiler;
+    if (BLOCKLIST_COMPILERS.includes(compiler)) {
+      return;
+    }
+
+    if (!(bucket in executionTime)) {
```

Comment:
Maybe using a general method to  add the object in the path if it does not exist, for instance


const ensurePath = (obj, ...keys) => {
  let current = obj;
  for (let i = 0; i < keys.length - 1; i++) {
    const key = keys[i];
    if (!current[key]) {
      current[key] = {};
    }
    current = current[key];
  }
  const lastKey = keys[keys.length - 1];
  if (!current[lastKey]) {
    current[lastKey] = [];
  }
  return current[lastKey];
};
const compilerArray = ensurePath(executionTime, bucket, workflowId, suite, compiler);

Reply:
Yeah, this sounds like a good idea. There is too much copy pasting there, so let me implement this